services/QuillLMS/app/controllers/api/v1/feedback_history_ratings_controller.rb

```diff
@@ -0,0 +1,19 @@
+class Api::V1::FeedbackHistoryRatingsController < ApplicationController
+  before_action :set_feedback_history_rating, only: [:show, :edit, :update, :destroy]
+  skip_before_action :verify_authenticity_token
+
+  def create_or_update
+    rating = FeedbackHistoryRating.find_or_create_by(feedback_history_rating_params)
```

Comment:
> We actually don't have info about who the user is easily accessible in the frontend - we could get it, but it would probably be easier to use `current_user` here instead of passing that information into the frontend just to pass it back.

👍

Reply:
> I also don't think this is actually going to work to update an existing one, because if for instance we already have a `FeedbackHistoryRating` with `{ user_id: 1, feedback_history_id: 1, rating: true }`, and then the user changes their mind and the argument here becomes `{ user_id: 1, feedback_history_id: 1, rating: false }`, it's always just going to create a new record. It seems to me like it would be easiest to do the lookup by user id and feedback history id and create or update based on that.

I'll break this out into separate controller actions.

Reply:
Alternatively, I think you could still use `find_or_create_by`, but only with `user_id` and `feedback_history_id` from the params.  Then update the `rating` on the object that you get from that call and save it.